contents/tutorials/test-frontend-feature-flags.md

```diff
@@ -14,32 +14,80 @@ Combining both testing and feature flags can be a bit tricky. Tests generally ch
 
 To do this, you need to mock the flags to access the other variations. This tutorial shows you how to do that by creating a React app with Jest tests, adding PostHog, then setting up tests that work with feature flags by mocking PostHog.
 
-## Creating a React app
+## Creating a Vite-React app and setup Vitest
 
-First, create a basic React app. Make sure [Node is installed](https://nodejs.dev/en/learn/how-to-install-nodejs/) then run `npx create-react-app` with the name of your app. We named ours `flag-test`.
+First, ensure [Node.js is installed](https://nodejs.dev/en/learn/how-to-install-nodejs/) (version 18.0 or newer). Then create a new React app with Vite: We named ours `flag-test`.
 
 ```bash
-npx create-react-app flag-test
+npm create vite@latest flag-test -- --template react
 ```
 
-After creating the app, in the newly created `flag-test` folder, a test is automatically created in `src/App.test.js` that looks like this:
+After creating the app, in the newly created `flag-test` folder, but by default the test framework isn't installed at the project yet, so let's add it.
+
+```bash
+cd flag-test
+npm add -D vitest @testing-library/react @testing-library/jest-dom
+```
+
+Since we're about to test `components` directly, we must create a file called `vitest.setup.js` at our project root importing the required dependency:
 
 ```js
-// src/App.test.js
-import { render, screen } from '@testing-library/react';
+// ./vitest.setup.js
+import '@testing-library/jest-dom'
+```
+
+With the dependency installed at our project, the next step is to let the `vite.config.js` know the new testing setup configurations by adding the `test` property:
+
+```js
+// ./vite.config.js
+import { defineConfig } from 'vite'
+import react from '@vitejs/plugin-react'
+
+export default defineConfig({ 
+  plugins: [react()],
+  test: {
+    globals: true,
+    environment: 'jsdom',
+    setupFiles: './vitest.setup.js'
+  },
+})
+```
+
+Now, we're ready to set up our first test. Since the **vite-react** already provided a sample component `src/App.jsx`, we can create a new file called `src/App.test.jsx` and start with this boilerplate:
```

Comment:
```suggestion
Now, we're ready to set up our first test. Since **vite-react** already provided a sample component `src/App.jsx`, we can create a new file called `src/App.test.jsx` and test it:
```